jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/status/CodeWhispererStatusBarManager.kt

```diff
@@ -31,6 +31,7 @@ class CodeWhispererStatusBarManager(private val project: Project) {
             ToolkitConnectionManagerListener.TOPIC,
             object : ToolkitConnectionManagerListener {
                 override fun activeConnectionChanged(newConnection: ToolkitConnection?) {
+                    if (project.isDisposed) return
```

Comment:
probably make `CodeWhispererStatusBarManager` disposable and attach itself to the `connect()` call so that the listener is disposed automatically

Reply:
gotcha